### ISSUE TITLE
understanding R^2 numbers

### ISSUE THREAD

**Issue by naarkhoo** [vegandevs/vegan#231]
I am using adonis for an analysis, where X is explained by three factors.
`x ~ f1 + f2 + f3`

If I understand correctly, adonis try to explain the variance of X by F1 and that is how the R^2 for F1 can be calculated. The left over from the variance, will be explained by F2 and similarly, the R^2 for F3 is calculated based on what ever F3 can explain the residuals from F1 and F2. In other words, the first factor takes as much as variance (variability) it corresponds to from x and what ever leaves, will be take by F2 and what ever left over will be taken by the third factor.

Here is the scenario, 
Let there are two models, 

`model 1) x ~ f1 + f2 + f3`
`model 2) x ~ f1 + f3 + f2`

I have noticed the R^2 for F2 in the second model is larger than its R^2 in the first model. I am surprised, because I expected to see a lower R^2 for F2 in the second model, since it has less chance to take more variability. However, still it is smaller than F2.

The table, below is what I see as an example

| Factor  | Model 1 | Model 2 |
| ------------- | ------------- | ------------- |
| F1  | 0.027  | 0.027 |
| F2  | 0.026  | 0.048 |
| F3  | 0.121  | 0.099 |

In other words, I am surprised to see F2  in the second model is larger than F2 in the first model even though it is ordered as the last factor.

**Comment by jarioksa** [comment]
This may be surprising, but life is. When the variables are correlated, their order will influence R^2 in the model. Positive and negative correlations will effect variables in different ways. The R^2 are based on sequential sums of squares, and the essential feature is that R^2 of both models sums up to 0.174.

**Comment by jarioksa** [comment]
In the following output
```r
Terms added sequentially (first to last)

           Df SumsOfSqs MeanSqs F.Model      R2 Pr(>F)    
Management  3    1.4686 0.48953  3.9548 0.34161  0.001 ***
A1          1    0.4409 0.44089  3.5618 0.10256  0.004 ** 
Moisture    3    0.9042 0.30139  2.4349 0.21032  0.008 ** 
Residuals  12    1.4854 0.12378         0.34551           
Total      19    4.2990                 1.00000        
```
The `R2` column is simply the `SumsOfSqs` column divided by its `Total` (last entry of the column). This indeed is order-dependent. In `adonis2` we use only the standard `anova` table output which lacks column `R2` and row `Total`. However, it allows the use of marginal tests which show the unique impact of each variable when added after all others:
```r
adonis2(formula = dune ~ Management + A1 + Moisture, data = dune.env, by = "margin")
           Df SumOfSqs      F Pr(>F)   
Management  3  0.90362 2.4334  0.013 * 
A1          1  0.19344 1.5627  0.184   
Moisture    3  0.90417 2.4349  0.010 **
Residual   12  1.48537                 
```
The `SumOfSqs` column provides the very same information that was used to calculate the `R2` except that we do not provide `Total`. We could very easily add this line, and fairly easily also the `R2` column. However, in sequential model (and in `adonis`) the `R2` values of terms add up to 1, but in marginal models they do not (in this case the marginal `R2` would add to 0.811). If you compare the two tables (`adonis` with sequential terms, `adonis2` with `by = "margin"`), you see that the last term (`Moisture`) has the same `SumOfSqs` in both cases. To get the marginal effects, you do not need to try all possible orders, but it is sufficient to see the effect when the term is the last in the table, and from these you can collect the `R2` values corresponding to the marginal effects. You may do this with `adonis` while we think if we should add the `R2` column to the `adonis2` output allowing direct analysis of marginal effects.

**Comment by jarioksa** [comment]
`adonis2` output has now a column for `R2` and is more similar to the classic `adonis` output. The `R2` column is directly found from adjacent `SumOfSqs` column, and when sum of squares are non-additive in marginal models, the `R2` neither sum up to 1. 

This change is in the `master` branch and will be released in the major release 2.5-0. I have no plans of cherry-picking this change to minor `2.4-z` releases.
